jans-cedarling/cedarling/src/common/policy_store.rs

```diff
@@ -5,155 +5,430 @@
  * Copyright (c) 2024, Gluu, Inc.
  */
 
-use std::collections::HashMap;
-
 use super::cedar_schema::CedarSchema;
+use base64::prelude::*;
+use cedar_policy::PolicyId;
+use serde::{de::Visitor, Deserialize, Deserializer};
+use std::{collections::HashMap, fmt};
 
-/// PolicyStoreMap it is a store for `PolicyStore` accessible by key.
-#[derive(Debug, serde::Deserialize)]
-pub(crate) struct PolicyStoreMap {
-    #[serde(flatten)]
-    pub policy_stores: HashMap<String, PolicyStore>,
-}
-
-/// PolicyStore contains all the data the Cedarling needs to verify JWT tokens and evaluate policies
+/// Represents the store of policies used for JWT validation and policy evaluation in Cedarling.
+///
+/// The `PolicyStore` contains the schema and a set of policies encoded in base64,
+/// which are parsed during deserialization.
 #[derive(Debug, Clone, serde::Deserialize)]
 pub struct PolicyStore {
-    pub schema: CedarSchema,
-    #[serde(deserialize_with = "deserialize_policy_set::parse")]
-    pub policies: cedar_policy::PolicySet,
+    /// The cedar version to use when parsing the schema and policies.
+    #[serde(deserialize_with = "check_cedar_version")]
+    #[allow(dead_code)]
+    pub cedar_version: String,
```

Comment:
Why not use structure with 3 fields, major, minor, patch? 

Or use crate [semver](https://docs.rs/semver/1.0.23/semver)

Reply:
replace custom version parsing with the `semver` crate here: [f07fcc2](https://github.com/JanssenProject/jans/pull/9910/commits/f07fcc2e45d6cb0e4e638f77ebf7f51632659042)